test/testGenVectorVc.cxx

```diff
@@ -0,0 +1,365 @@
+#include <random>
+#include <vector>
+#include <iostream>
+#include <string>
+#include <typeinfo>
+#include <cmath>
+
+// note scale here is > 1 as SIMD and scalar floating point calculations not
+// expected to be bit wise identical
+int compare(double v1, double v2, const std::string &name = "", double scale = 1000.0)
+{
+   //  ntest = ntest + 1;
+
+   // numerical double limit for epsilon
+   const double eps     = scale * std::numeric_limits<double>::epsilon();
+   int          iret    = 0;
+   double       delta   = v2 - v1;
+   double       d       = 0;
+   if (delta < 0) delta = -delta;
+   if (v1 == 0 || v2 == 0) {
+      if (delta > eps) {
+         iret = 1;
+      }
+   }
+   // skip case v1 or v2 is infinity
+   else {
+      d = v1;
+
+      if (v1 < 0) d = -d;
+      // add also case when delta is small by default
+      if (delta / d > eps && delta > eps) iret = 1;
+   }
+
+   if (iret != 0) {
+      int pr = std::cout.precision(18);
+      std::cout << "\nDiscrepancy in " << name << "() : " << v1 << " != " << v2 << " discr = " << int(delta / d / eps)
+                << "   (Allowed discrepancy is " << eps << ")\n";
+      std::cout.precision(pr);
+   }
+   return iret;
+}
+
+// randomn generator
+static std::default_random_engine gen;
+// Distributions for each member
+static std::uniform_real_distribution<double> p_x(-800, 800), p_y(-600, 600), p_z(10000, 10500);
+static std::uniform_real_distribution<double> d_x(-0.2, 0.2), d_y(-0.1, 0.1), d_z(0.95, 0.99);
+static std::uniform_real_distribution<double> c_x(3100, 3200), c_y(10, 15), c_z(3200, 3300);
+static std::uniform_real_distribution<double> r_rad(8500, 8600);
+static std::uniform_real_distribution<double> p0(-0.002, 0.002), p1(-0.2, 0.2), p2(0.97, 0.99), p3(-1300, 1300);
+
+template <typename POINT, typename VECTOR, typename PLANE, typename FTYPE>
+class Data {
+public:
+   typedef std::vector<Data, Vc::Allocator<Data>> Vector;
+
+public:
+   POINT  position;
+   VECTOR direction;
+   POINT  CoC;
+   PLANE  plane;
+   FTYPE  radius{0};
+
+public:
+   template <typename INDATA>
+   Data(const INDATA &ind)
+      : position(ind.position.x(), ind.position.y(), ind.position.z()),
+        direction(ind.direction.x(), ind.direction.y(), ind.direction.z()), CoC(ind.CoC.x(), ind.CoC.y(), ind.CoC.z()),
+        plane(ind.plane.A(), ind.plane.B(), ind.plane.C(), ind.plane.D()), radius(ind.radius)
+   {
+   }
+   Data()
+      : position(p_x(gen), p_y(gen), p_z(gen)), direction(d_x(gen), d_y(gen), d_z(gen)),
+        CoC(c_x(gen), c_y(gen), c_z(gen)), plane(p0(gen), p1(gen), p2(gen), p3(gen)), radius(r_rad(gen))
+   {
+   }
+};
+
+template <typename INDATA, typename OUTDATA>
+void fill(const INDATA &in, OUTDATA &out)
+{
+   out.clear();
+   out.reserve(in.size());
+   for (const auto &i : in) {
+      out.emplace_back(i);
+   }
+}
+
+template <typename POINT, typename VECTOR, typename FTYPE>
+inline
+   typename std::enable_if<std::is_arithmetic<typename POINT::Scalar>::value &&
```

Comment:
@cjones051073 it seems that we omitted transforming this code here as a default template parameter.